### DISCUSSION TITLE
umqtt.simple documentation

### DISCUSSION THREAD

**Discussion by smhodge42** [micropython/micropython#13659] [ESP32]
Is there "official" documentation for this package (in my case, for use on ESP32)?  There seems to be some on websites like "mpython" or "pypi" but I'm not sure that's the version I get by importing the package (in firmware from the Micropython website) onto an ESP32-S3, which I've done and been able to kludge together from examples scattered around the web and a local test broker on a raspberry pi.

But a search on the official Micropython "documents" comes up empty.  I've found the source code on github, but nowhere can I find documentation on how to use each method, detailing usage, arguments, returns, etc., like you get, say, if you look up "I2C" or "SPI" for a specific Micropython port on the Micropython website.

**Comment by andrewleech** [comment]
Umqtt.simple is installed from micropython-lib, these packages generally aren't covered on the official docs page as that's generated from the main micropython repo.

There's examples in micropython-lib however docs are pretty patchy.
